app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -147,6 +149,9 @@ class HomeFragment : Fragment() {
 
         lifecycle.addObserver(sessionObserver)
 
+        if (finishOnboarding)
+            onboarding.finish()
```

Comment:
I wonder if calling `hideOnboarding()` is more appropriate. Functionally, I think it's the same, however.

I suspect this is moot if we do my suggestion above.